docs/specification/draft/client/user_interaction.mdx

```diff
@@ -0,0 +1,355 @@
+`notifications/cancelled` message to the client, as described in [Cancellation](/specification/draft/basic/utilities/cancellation).
+The MCP client **SHOULD NOT** send a response.
+
+
+## Security Considerations
+
+### Handling Sensitive Information
+
+MCP servers **MUST** use the `ua` interaction type for sensitive information, such as
+auth credentials or authorization grants. The MCP client **MUST NOT** send sensitive
+information to the MCP server in any other way.
+
+### User Consent
+
+The MCP client **SHOULD** clearly communicate the purpose of the interaction to the end-user,
+and provide a way for the end-user to consent to the interaction before continuing.
+
+### Identifying the User
+
+The MCP server **MUST NOT** rely on input from the client or end-user to identify the
+end-user, as this untrusted input can be forged and lead to impersonation attacks.
+
+Instead, the MCP server **SHOULD** identify the end-user in trusted (server) code. For
+example:
+
+- Implementations using an HTTP transport and [MCP authorization](/specification/draft/basic/authorization)
+  **MAY** use the subject claim (`sub`) from the OAuth 2.1 Access Token as the user
+  identifier, if present.
+- Implementations using an stdio transport **MAY** acquire the user identifier from the
+  environment.
+
+### Server-side Request Forgery
+
+With the `ua` interaction type, the MCP client is expected to make a request via a user agent
+based on the value of the `url` parameter in the interaction object. Since MCP enables clients
+to interoperate with MCP servers it has no prior knowledge of, this opens a risk for Server-Side
+Request Forgery (SSRF) attacks by malicious MCP servers. MCP clients **SHOULD** take appropriate
+precautions against SSRF attacks, such as blocking requests to internal IP address ranges and
+requiring TLS.
+
+Additionally, the MCP client **SHOULD** validate the `url` parameter in the interaction object and
+provide sufficient context to the user about the request that will be made, before it is opened in
+a User Agent.
+
+Further recommendations can be found in the Open Worldwide Application Security
+Project (OWASP) [SSRF Prevention Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.html).
+
+### Phishing
+
+The `ua` interaction type may be used in a way where the `url` parameter in the
+interaction object points to a URL on a domain that is not the MCP server's domain. For
+example, the URL may be an authorization URL of a third-party service that the MCP server
+needs the user to authorize access to. If that third-party service is malicious or
+compromised, it could trick the user into revealing their account credentials or other
+sensitive information.
+
+To mitigate this risk, MCP clients **SHOULD** take appropriate precautions, such as verifying the
+authenticity of the URL and clearly presenting the destination of the request to the user.
```

Comment:
This also needs a lot more details as to what "authenticity" here means and what information the client will reveal to the URL presented.